### ISSUE TITLE
Missing upload tree parent for upload 

### ISSUE THREAD

**Issue by jb68** [fossology/fossology#1697]
### Description
I am trying to start a new scan by uploading from URL (using the web interface ) If I go to the job 'Geeky Scan Details' and click on 'Browse upload' I get 
Missing upload tree parent for upload 

#### How to reproduce
Delete some older uploads and folders using api/v1
Try to upload a new scan, you will get an old upload ID.

### Versions

3.7 official docker 

### Logs
```
2020-04-21 18:31:34 reuser [0] :: JOB[31385].reuser[202.localhost]: "PHP Fatal error:  Uncaught TypeError: Argument 1 passed to Fossology\Lib\Dao\ClearingDao::getFileClearingsFolder() must be an instance of Fossology\Lib\Data\Tree\ItemTreeBounds, boolean given, called in /usr/local/share/fossology/reuser/agent/ReuserAgent.php on line 207 and defined in /usr/local/share/fossology/lib/php/Dao/ClearingDao.php:174"
2020-04-21 18:31:34 reuser [0] :: JOB[31385].reuser[202.localhost]: "Stack trace:"
2020-04-21 18:31:34 reuser [0] :: JOB[31385].reuser[202.localhost]: "#0 /usr/local/share/fossology/reuser/agent/ReuserAgent.php(207): Fossology\Lib\Dao\ClearingDao->getFileClearingsFolder(false, '3')"
2020-04-21 18:31:34 reuser [0] :: JOB[31385].reuser[202.localhost]: "#1 /usr/local/share/fossology/reuser/agent/ReuserAgent.php(114): Fossology\Reuser\ReuserAgent->processEnhancedUploadReuse(false, Object(Fossology\Lib\Data\Tree\ItemTreeBounds), '3')"
2020-04-21 18:31:34 reuser [0] :: JOB[31385].reuser[202.localhost]: "#2 /usr/local/share/fossology/lib/php/Agent/Agent.php(338): Fossology\Reuser\ReuserAgent->processUploadId(1447)"
2020-04-21 18:31:34 reuser [0] :: JOB[31385].reuser[202.localhost]: "#3 /usr/local/share/fossology/reuser/agent/reuser.php(59): Fossology\Lib\Agent\Agent->run_scheduler_event_loop()"
2020-04-21 18:31:34 reuser [0] :: JOB[31385].reuser[202.localhost]: "#4 /usr/local/lib/fossology/fo_wrapper(77): require('/usr/local/shar...')"
2020-04-21 18:31:34 reuser [0] :: JOB[31385].reuser[202.localhost]: "#5 {main}"
2020-04-21 18:31:34 reuser [0] :: JOB[31385].reuser[202.localhost]: "  thrown in /usr/local/share/fossology/lib/php/Dao/ClearingDao.php on line 174"
2020-04-21 18:31:34 reuser [0] :: JOB[31385].reuser[202.localhost]: agent failed, code: 255
```

And:
```
2020-04-21 18:31:34 decider [0] :: JOB[31384].decider[206.localhost]: "Caught exception while processing uploadId=1447: Missing upload tree parent for upload#0 /usr/local/share/fossology/decider/agent/BulkReuser.php(65): Fossology\Lib\Dao\UploadDao->getUploadParent(1447)"
2020-04-21 18:31:34 decider [0] :: JOB[31384].decider[206.localhost]: "#1 /usr/local/share/fossology/decider/agent/DeciderAgent.php(161): Fossology\Decider\BulkReuser->rerunBulkAndDeciderOnUpload(1447, '3', '3', '4510', 0)"
2020-04-21 18:31:34 decider [0] :: JOB[31384].decider[206.localhost]: "#2 /usr/local/share/fossology/lib/php/Agent/Agent.php(338): Fossology\Decider\DeciderAgent->processUploadId(1447)"
2020-04-21 18:31:34 decider [0] :: JOB[31384].decider[206.localhost]: "#3 /usr/local/share/fossology/decider/agent/decider.php(33): Fossology\Lib\Agent\Agent->run_scheduler_event_loop()"
2020-04-21 18:31:34 decider [0] :: JOB[31384].decider[206.localhost]: "#4 /usr/local/lib/fossology/fo_wrapper(77): require('/usr/local/shar...')"
2020-04-21 18:31:34 decider [0] :: JOB[31384].decider[206.localhost]: "#5 {main}agent failed on uploadId=1447#0  Fossology\Lib\Agent\Agent->bail() called at [/usr/local/share/fossology/lib/php/Agent/Agent.php:349]"
2020-04-21 18:31:34 decider [0] :: JOB[31384].decider[206.localhost]: "#1  Fossology\Lib\Agent\Agent->run_scheduler_event_loop() called at [/usr/local/share/fossology/decider/agent/decider.php:33]"
2020-04-21 18:31:34 decider [0] :: JOB[31384].decider[206.localhost]: "#2  require(/usr/local/share/fossology/decider/agent/decider.php) called at [/usr/local/lib/fossology/fo_wrapper:77]"
2020-04-21 18:31:34 decider [0] :: JOB[31384].decider[206.localhost]: agent failed with error code 1
2020-04-21 18:31:34 decider [0] :: JOB[31384].decider[206.localhost]: agent failed, code: 0
```

**Comment by mcjaeger** [comment]
@jb68 thanks for reporting to us. would you be able to share the REST call with us? We re suspecting that the setting for the reuse agent is invalid? Maybe it is possible to create a a request with an an invalid upload id when enabling reuse?

**Comment by jb68** [comment]
I've tried from the web interface with and without the reuse, same result. 
From Jenkins is bit more complicated.. 
First I am copy the file from an url with docker:
`docker exec "$DOCKER_ENCAPS_NAME" cp2foss -n $PROJECT_NAME -f $PROJECT_PATH -P 1 $ARCHIVE_FILE > result`

this returns 
`upload_pk=$(cat result | awk '/UploadPk is:/ {print $3}' | sed "s|'||g")
folder_pk=$(cat result | awk '/FolderPk is:/ {print $3}' | sed "s|'||g")`

Then I check the job status:
`for i in {1..150}; do
  FOSS_JOB_STATUS=$(curl -k -s -X GET http://${FOSSOLOGY_SERVER}/repo/api/v1/jobs?upload=$upload_pk \
    -H "Authorization: ${FOSSOLOGY_TOKEN}" \
    -H "accept: application/json")
  FOSS_JOB_STATUS=$(echo $FOSS_JOB_STATUS | jq ".[] | select (.name==\"wget\")" | jq -sr .[-1].status)` 

This part fails and I discovered is related with https://github.com/fossology/fossology/issues/1684 because it shows that the returned $upload_pk has been also used by older jobs ( that I deleted )

As an interesting fact by retrying multiple times the upload_pk incremented to bigger values so it doesn't do it any more ( at least is usable again )

**Comment by ag4ums** [comment]
hello @jb68 , just like to understand how you are deploying and deleting the packages in fossology..... 
1) are you deploying a new container every time while uploading fresh packages and 
2) when you are deleting the uploaded packages from that container. 
could you please provide us bit more insight....

**Comment by jb68** [comment]
Hi,
I didn't deleted any packages until now but now the HD is close to full so I decided to delete some old packages to make space. I created a script that figure out:
- list all uploads `SELECT upload_pk, upload_filename FROM upload`
- Determine how old is a package and if a package, and if there is a newer package in same folder etc 

If needs to be deleted I use: 
`local response=$(curl -s -X DELETE "http://${FOSSOLOGY_SERVER}/repo/api/v1/uploads/$upload_pk" \
            -H  "accept: application/json" \
            -H "Authorization: ${FOSSOLOGY_TOKEN}")
        local result=$(fossology_check_response 202 "message" "$response")`

Then I check if folder is empty with `SELECT folder_pk,parent FROM folderlist WHERE folder_pk NOT IN (SELECT parent FROM folderlist)` 
and delete it with:
`local response=$(curl -s -X DELETE "http://${FOSSOLOGY_SERVER}/repo/api/v1/folders/$folder_pk" \
            -H  "accept: application/json" \
            -H "Authorization: ${FOSSOLOGY_TOKEN}")
        local result=$(fossology_check_response 202 "message" "$response")`

After this I get `Missing upload tree parent for upload ` on new uploads that are suppose to go in existent folders. The error comes from both web interface as well as our automation

**Comment by GMishx** [comment]
Should be fixed by #1707

**Comment by shaheemazmalmmd** [comment]
@jb68 not able to reproduce the issue after tests on latest master. i think it was fixed by #1707 . please re open in case the issue persist on your side.
